src/Microsoft.AspNet.Mvc.Core/FilterActionInvoker.cs

```diff
@@ -150,6 +153,30 @@ private IFilter[] GetFilters()
             return filterProviderContext.Results.Select(item => item.Filter).Where(filter => filter != null).ToArray();
         }
 
+        private async Task InvokeAllAuthenticationFiltersAsync()
```

Comment:
That was not the case with the `HostAuthenticationFilter` built in Web API OWIN, but have you considered storing the existing `context.User` reference and restoring it when MVC stops processing the request, so the ambient user doesn't change for the middleware higher in the pipeline?

(that was a concern @davidfowl had IIRC)